analyzers/tests/SonarAnalyzer.UnitTest/MetadataReferences/NuGetMetadataFactory.cs

```diff
@@ -29,15 +29,18 @@
 using Microsoft.CodeAnalysis;
 using NuGet;
 
-namespace SonarAnalyzer.UnitTest.TestFramework
+namespace SonarAnalyzer.UnitTest.MetadataReferences
```

Comment:
Cleanup to move it together with `MetadataReferenceFactory` and `NuGetMetadataReference`. And to remove it from default intellisense, because finding `NuGetMetadataReference` was hard and annoying when only NugetMetadataFactory appeared by default.